src/ProjectService.Business/Commands/Project/EditProjectCommand.cs

```diff
@@ -91,23 +95,22 @@ public OperationResultResponse<bool> Execute(Guid projectId, JsonPatchDocument<E
         {
             _validator.ValidateAndThrowCustom(request);
 
-            GetProjectFilter filter = new GetProjectFilter { ProjectId = projectId };
-
-            DbProject dbProject = _repository.Get(filter);
+            DbProject dbProject = _projectRepository.Get(new GetProjectFilter { ProjectId = projectId, IncludeUsers = true });
 
             OperationResultResponse<bool> response = new();
+            Guid userId = _httpContextAccessor.HttpContext.GetUserId();
 
             if (!_accessValidator.IsAdmin() &&
-                GetDepartment(dbProject.DepartmentId, response.Errors).DirectorUserId !=
-                _httpContextAccessor.HttpContext.GetUserId())
+                GetDepartment(dbProject.DepartmentId, response.Errors)?.DirectorUserId != userId &&
+                !dbProject.Users.Any(user => user.UserId == userId && user.Role == (int)ProjectUserRoleType.Manager))
```

Comment:
it would be better if you put the check on the manager before checking on the director of the department. I advise you to do this because the request for the department director is processed through the broker and it would be better to avoid unnecessary requests to the broker

Reply:
done